app/controllers/v2/LiabilityController.scala

```diff
@@ -65,4 +74,39 @@ class LiabilityController @Inject()(
       Some(AttachmentsTabViewModel(liabilityCase.reference, applicantFiles, letter, nonApplicantFiles))
     }
   }
+
+
+  def addNote(reference: String): Action[AnyContent] = (verify.authenticated andThen
+    verify.casePermissions(reference) andThen verify.mustHave(Permission.ADD_NOTE)).async { implicit request =>
+    def onError: Form[ActivityFormData] => Future[Result] = errorForm => {
+      val liabilityCase: Case = request.`case`
```

Comment:
much of this code is common with the main get function - can you factor it out?

Reply:
we've moved it to a separate function